js/dataframe/shared/Table.svelte

```diff
@@ -159,7 +159,7 @@
 
 	$: if (!dequal(values, old_val)) {
 		data = process_data(values as (string | number)[][]);
-		old_val = values as (string | number)[][];
+		old_val = JSON.parse(JSON.stringify(values)) as (string | number)[][];
```

Comment:
This is the main fix for the issues -- doing a deep copy here. Everything else is just cleanup.